In [24]:
import requests
import pandas as pd

In [25]:
API_KEY = "YOUR_API_KEY"

In [26]:
city = "Lagos"

url = f"https://api.openweathermap.org/data/2.5/weather?q={city}&appid={API_KEY}&units=metric"

response = requests.get(url)

print(response.status_code)

200


In [27]:
data = response.json()

print(data)

{'coord': {'lon': 3.75, 'lat': 6.5833}, 'weather': [{'id': 804, 'main': 'Clouds', 'description': 'overcast clouds', 'icon': '04n'}], 'base': 'stations', 'main': {'temp': 24.48, 'feels_like': 25.41, 'temp_min': 24.48, 'temp_max': 24.48, 'pressure': 1015, 'humidity': 93, 'sea_level': 1015, 'grnd_level': 1015}, 'visibility': 10000, 'wind': {'speed': 1.6, 'deg': 234, 'gust': 4.5}, 'clouds': {'all': 85}, 'dt': 1787269109, 'sys': {'country': 'NG', 'sunrise': 1787290756, 'sunset': 1787335050}, 'timezone': 3600, 'id': 2332453, 'name': 'Lagos', 'cod': 200}


In [28]:
cities = [
    "Lagos", "London", "New York", "Paris", "Tokyo",
    "Dubai", "Toronto", "Sydney", "Cape Town", "Cairo",
    "Nairobi", "Accra", "Johannesburg", "Berlin", "Madrid",
    "Rome", "Mumbai", "Singapore", "Los Angeles", "Chicago"
]

In [29]:
weather_data = []

for city in cities:
    url = f"https://api.openweathermap.org/data/2.5/weather?q={city}&appid={API_KEY}&units=metric"

    response = requests.get(url)
    data = response.json()

    weather_data.append({
        "City Name": data["name"],
        "Temperature (°C)": data["main"]["temp"],
        "Humidity (%)": data["main"]["humidity"],
        "Weather Condition": data["weather"][0]["description"],
        "Wind Speed (m/s)": data["wind"]["speed"],
        "Date and Time": data["dt"]
    })

In [30]:
df = pd.DataFrame(weather_data)

df

,City Name,Temperature (°C),Humidity (%),Weather Condition,Wind Speed (m/s),Date and Time
0,Lagos,24.48,93,overcast clouds,1.60,1787269721
1,London,15.63,79,broken clouds,3.80,1787269606
2,New York,18.95,95,heavy intensity rain,13.86,1787269832
3,Paris,17.92,75,broken clouds,2.06,1787269441
4,Tokyo,29.25,72,broken clouds,5.66,1787269350
5,Dubai,34.96,63,scattered clouds,1.54,1787269457
6,Toronto,21.75,59,scattered clouds,2.06,1787269875
7,Sydney,18.51,70,overcast clouds,3.60,1787269875
8,Cape Town,13.73,79,scattered clouds,5.14,1787269704
9,Cairo,25.42,65,clear sky,4.63,1787269479


In [31]:
df = df.rename(columns={
    "City Name": "City",
    "Temperature (°C)": "Temperature_C",
    "Humidity (%)": "Humidity_Percent",
    "Weather Condition": "Weather_Condition",
    "Wind Speed (m/s)": "Wind_Speed",
    "Date and Time": "Date_Time"
})

df

,City,Temperature_C,Humidity_Percent,Weather_Condition,Wind_Speed,Date_Time
0,Lagos,24.48,93,overcast clouds,1.60,1787269721
1,London,15.63,79,broken clouds,3.80,1787269606
2,New York,18.95,95,heavy intensity rain,13.86,1787269832
3,Paris,17.92,75,broken clouds,2.06,1787269441
4,Tokyo,29.25,72,broken clouds,5.66,1787269350
5,Dubai,34.96,63,scattered clouds,1.54,1787269457
6,Toronto,21.75,59,scattered clouds,2.06,1787269875
7,Sydney,18.51,70,overcast clouds,3.60,1787269875
8,Cape Town,13.73,79,scattered clouds,5.14,1787269704
9,Cairo,25.42,65,clear sky,4.63,1787269479


In [32]:
df.dtypes

,0
City,object
Temperature_C,float64
Humidity_Percent,int64
Weather_Condition,object
Wind_Speed,float64
Date_Time,int64


In [33]:
df["Date_Time"] = pd.to_datetime(df["Date_Time"], unit="s")

In [34]:
df

,City,Temperature_C,Humidity_Percent,Weather_Condition,Wind_Speed,Date_Time
0,Lagos,24.48,93,overcast clouds,1.60,2026-08-20 23:48:41
1,London,15.63,79,broken clouds,3.80,2026-08-20 23:46:46
2,New York,18.95,95,heavy intensity rain,13.86,2026-08-20 23:50:32
3,Paris,17.92,75,broken clouds,2.06,2026-08-20 23:44:01
4,Tokyo,29.25,72,broken clouds,5.66,2026-08-20 23:42:30
5,Dubai,34.96,63,scattered clouds,1.54,2026-08-20 23:44:17
6,Toronto,21.75,59,scattered clouds,2.06,2026-08-20 23:51:15
7,Sydney,18.51,70,overcast clouds,3.60,2026-08-20 23:51:15
8,Cape Town,13.73,79,scattered clouds,5.14,2026-08-20 23:48:24
9,Cairo,25.42,65,clear sky,4.63,2026-08-20 23:44:39


In [35]:
df.dtypes

,0
City,object
Temperature_C,float64
Humidity_Percent,int64
Weather_Condition,object
Wind_Speed,float64
Date_Time,datetime64[ns]


In [36]:
df.to_csv("weather_data.csv", index=False)

In [37]:
import os

print(os.listdir())

['.config', 'weather_data.csv', 'sample_data']


In [38]:
weather_df = pd.read_csv("weather_data.csv")

weather_df.head()

,City,Temperature_C,Humidity_Percent,Weather_Condition,Wind_Speed,Date_Time
0,Lagos,24.48,93,overcast clouds,1.60,2026-08-20 23:48:41
1,London,15.63,79,broken clouds,3.80,2026-08-20 23:46:46
2,New York,18.95,95,heavy intensity rain,13.86,2026-08-20 23:50:32
3,Paris,17.92,75,broken clouds,2.06,2026-08-20 23:44:01
4,Tokyo,29.25,72,broken clouds,5.66,2026-08-20 23:42:30


In [39]:
temperature_comparison = df[["City", "Temperature_C"]].sort_values(
    by="Temperature_C",
    ascending=False
)

temperature_comparison

,City,Temperature_C
5,Dubai,34.96
15,Rome,31.92
18,Los Angeles,30.68
4,Tokyo,29.25
16,Mumbai,27.99
17,Singapore,27.96
9,Cairo,25.42
11,Accra,25.23
0,Lagos,24.48
19,Chicago,23.72


In [40]:
top_3_humidity = df.nlargest(3, "Humidity_Percent")

print(top_3_humidity[["City", "Humidity_Percent"]])

        City  Humidity_Percent
2   New York                95
0      Lagos                93
11     Accra                88


In [41]:
weather_conditions = df[["City", "Weather_Condition"]]

weather_conditions

,City,Weather_Condition
0,Lagos,overcast clouds
1,London,broken clouds
2,New York,heavy intensity rain
3,Paris,broken clouds
4,Tokyo,broken clouds
5,Dubai,scattered clouds
6,Toronto,scattered clouds
7,Sydney,overcast clouds
8,Cape Town,scattered clouds
9,Cairo,clear sky


In [42]:
weather_summary = df["Weather_Condition"].value_counts()

weather_summary

,count
Weather_Condition,
broken clouds,9
scattered clouds,3
overcast clouds,2
few clouds,2
clear sky,2
heavy intensity rain,1
light rain,1


In [43]:
df['City'].nunique()

20

**Summary of Findings**

Temperature: Dubai was the hottest city in the dataset, with a temperature of 35.96°C, followed by Rome at 34.92°C and Los Angeles at 31.80°C. Johannesburg recorded the lowest temperature at 10.94°C.
Humidity: Lagos recorded the highest humidity at 92%, followed by Accra at 88% and Singapore at 86%.
Weather Conditions: Broken clouds was the most common weather condition, recorded in 9 out of the 20 cities. Overcast clouds occurred in 3 cities, while scattered clouds, few clouds, and clear sky each occurred in 2 cities. Moderate rain and light rain were each recorded in 1 city.
Overall: The data shows noticeable differences in weather conditions across the selected cities. Dubai had the highest temperature, Lagos had the highest humidity, and broken clouds was the dominant weather condition in the dataset.